src/pbkdf2.rs

```diff
@@ -150,9 +149,39 @@ pub fn derive(prf: &'static PRF, iterations: usize, salt: &[u8], secret: &[u8],
     // Clear |out|.
     polyfill::slice::fill(out, 0);
 
+    let mut idx: u32 = 0;
+
+    for chunk in out.chunks_mut(output_len) {
+        idx += 1;
+        let chunk_len = chunk.len();
+
+        if chunk_len == output_len {
+            derive_block(&secret, iterations, salt, idx, chunk);
+        } else {
+            let mut tmp = Vec::with_capacity(output_len);
+            for _ in 0..output_len {
+                tmp.push(0u8);
+            }
```

Comment:
Considering [this](https://github.com/briansmith/ring/pull/252/files#r71096506):

```rust
let tmp = [0; output_len];
```